tool/grind.dart

```diff
@@ -3,24 +3,120 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:convert';
 import 'dart:io' hide ProcessException;
 
-import 'package:dartdoc/dartdoc.dart' show defaultOutDir;
 import 'package:dartdoc/src/io_utils.dart';
 import 'package:grinder/grinder.dart';
-import 'package:html/dom.dart';
-import 'package:html/parser.dart' show parse;
 import 'package:path/path.dart' as path;
 import 'package:yaml/yaml.dart' as yaml;
 
 main([List<String> args]) => grind(args);
 
-final Directory docsDir =
-    new Directory(path.join('${Directory.systemTemp.path}', defaultOutDir));
+final Directory dartdocDocsDir = Directory.systemTemp.createTempSync('dartdoc');
+final Directory sdkDocsDir = Directory.systemTemp.createTempSync('sdkdocs');
+final Directory flutterDir = Directory.systemTemp.createTempSync('flutter');
+final Directory flutterDirDevTools =
+    new Directory(path.join(flutterDir.path, 'dev', 'tools'));
+final RegExp quotables = new RegExp(r'[ "\r\n\$]');
+
+// from flutter:dev/tools/dartdoc.dart, modified
+void _printStream(Stream<List<int>> stream, Stdout output,
+    {String prefix: ''}) {
+  assert(prefix != null);
+  stream
+      .transform(UTF8.decoder)
+      .transform(const LineSplitter())
+      .listen((String line) {
+    output.write('$prefix$line'.trim());
+    output.write('\n');
+  });
+}
+
+/// Creates a throwaway pub cache and returns the environment variables
+/// necessary to use it.
+Map<String, String> _createThrowawayPubCache() {
+  final Directory pubCache = Directory.systemTemp.createTempSync('pubcache');
+  final Directory pubCacheBin = new Directory(path.join(pubCache.path, 'bin'));
+  pubCacheBin.createSync();
+  return new Map.fromIterables([
+    'PUB_CACHE',
+    'PATH'
+  ], [
+    pubCache.path,
+    [pubCacheBin.path, Platform.environment['PATH']].join(':')
+  ]);
+}
+
+class _SubprocessLauncher {
+  final String context;
+  Map<String, String> _environment;
+
+  Map<String, String> get environment => _environment;
+
+  String get prefix => context.isNotEmpty ? '$context: ' : '';
+
+  _SubprocessLauncher(this.context, [Map<String, String> environment]) {
+    if (environment == null) this._environment = new Map();
+  }
+
+  /// A wrapper around start/await process.exitCode that will display the
+  /// output of the executable continuously and fail on non-zero exit codes.
+  /// Makes running programs in grinder similar to set -ex for bash, even on
+  /// Windows (though some of the bashisms will no longer make sense).
+  /// TODO(jcollins-g): move this to grinder?
+  Future runStreamed(String executable, List<String> arguments,
+      {String workingDirectory}) async {
+    stderr.write('$prefix+ ');
```

Comment:
you want stderr here?

Reply:
Yes.  Modeled after set -x in bash, which echos the commands used to stderr.